examples/aes-ccm/src/ccm_mode.rs

```diff
@@ -0,0 +1,197 @@
+use hacspec_lib::*;
+use hacspec_aes::*;
+
+fn format_func(a: &ByteSeq, n: &ByteSeq, p: &ByteSeq, t: u8, alen: u64, nlen: u8, plen: u64) -> ByteSeq {
+    let mut r = 0u64;
+    let mut tmp = 10u64;
+
+    if alen < 0x800000u64 {
+        tmp = 2u64;
+    } else {
+        if alen < 0x100000000u64 {
+            tmp = 6u64;
+        }
+    }
+
+    r = r + ((tmp+alen+15u64)/16u64)+((plen+15u64)/16u64); // ceiling operation used
+    let mut b = ByteSeq::new((16u64*(r+1u64)) as usize);
+
+    // creation of b(0)
+    let qlen: u8 = 15u8 - nlen;
+    let mut flags: u8 = 0u8;
+
+    if alen > 0u64 {
+        flags = 0x40u8;
+    }
+
+    flags = flags | (((t-2u8)/2u8) << 3);
+    flags = flags | (qlen-1u8);
+    b[0] = U8(flags);
+
+    for i in 0..(nlen as usize) {
+        let tmp2 = n.get_exact_chunk(1, i);
+        b = b.set_exact_chunk(1, i+1, &tmp2);
+    }
+
+    let andy: u64 = 255u64; // 0xFF
+    let mut copy: u64 = plen;
```

Comment:
`copy` and `copy2` aren't great variable names. Please give them proper names so it's easier to understand what the variables are doing.

Reply:
Yup, I think I have used `copy`, `tmp` and `smth` in my code. I'll rename them.